app/Model/Color.php

```diff
@@ -100,6 +107,15 @@ class Color extends Base
     );
 
     /**
+     * Colors used in the app, since defaults can be overriden in the database.
+     * Each color also contains bool 'is_usable' array key.
+     *
+     * @access private
+     * @var array
+     */
+    private $colors = null;
```

Comment:
You can default to array() instead of null so php knows the type at variable creation.